oslabs-beta/reactime
# Changes to Tree class

## Description:
- Changes npm snapshot tree to keep separate states for hooks and setState, fixes component hierarchy, and adds performance data to snapshots.

## Changes I Made:
- Changed npm snapshot tree to keep separate states for hooks and setState. Now hooks state is in state: {hooksState: {}}.
- Kept separate indexes for hooks state changes in ComponentActionsRecord.
- createTree now properly creates component hierarchy.
- Added performance data to snapshots in componentData key of Tree class.

## How to Test:
- Start a react application with npm package installed, check new data in extension tree chart.
